google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java

```diff
@@ -457,28 +469,64 @@ void write(LogEntry entry, WriteOption... options) {
           reportError(null, ex, ErrorManager.FLUSH_FAILURE);
         }
         break;
+
       case ASYNC:
       default:
-        ApiFutures.addCallback(getLogging().writeAsync(entryList, options), new ApiFutureCallback<Void>() {
-          @Override
-          public void onSuccess(Void v) {}
-
-          @Override
-          public void onFailure(Throwable t) {
-            if (t instanceof Exception) {
-              reportError(null, (Exception) t, ErrorManager.FLUSH_FAILURE);
-            } else {
-              reportError(null, new Exception(t), ErrorManager.FLUSH_FAILURE);
-            }
-          }
-        });
+        final ApiFuture<Void> writeFuture = getLogging().writeAsync(entryList, options);
+        writeLock.lock();
+        try {
+          pendingWrites.add(writeFuture);
+        } finally {
+          writeLock.unlock();
+        }
```

Comment:
Done. I made `writeLock` an Object and used `synchronized(writeLock)`. We can also just use `synchronized(pendingWrites)`. Do you have an opinion?